aws/resource_aws_ec2_fleet.go

```diff
@@ -0,0 +1,839 @@
+								if defaultTargetCapacityTypeO.(string) != ec2.DefaultTargetCapacityTypeOnDemand {
+									return false
+								}
+								// Show difference if existing state reflects different total capacity
+								oldInt, err := strconv.Atoi(old)
+								if err != nil {
+									log.Printf("[WARN] %s DiffSuppressFunc error converting %s to integer: %s", k, old, err)
+									return false
+								}
+								totalTargetCapacityO, _ := d.GetChange("target_capacity_specification.0.total_target_capacity")
+								if oldInt != totalTargetCapacityO.(int) {
+									return false
+								}
+								return true
+							},
+						},
+						"spot_target_capacity": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								// Show difference for new resources
+								if d.Id() == "" {
+									return false
+								}
+								// Show difference if value is configured
+								if new != "0" {
+									return false
+								}
+								// Show difference if existing state reflects different default type
+								defaultTargetCapacityTypeO, _ := d.GetChange("target_capacity_specification.0.default_target_capacity_type")
+								if defaultTargetCapacityTypeO.(string) != ec2.DefaultTargetCapacityTypeSpot {
+									return false
+								}
+								// Show difference if existing state reflects different total capacity
+								oldInt, err := strconv.Atoi(old)
+								if err != nil {
+									log.Printf("[WARN] %s DiffSuppressFunc error converting %s to integer: %s", k, old, err)
+									return false
+								}
+								totalTargetCapacityO, _ := d.GetChange("target_capacity_specification.0.total_target_capacity")
+								if oldInt != totalTargetCapacityO.(int) {
+									return false
+								}
+								return true
+							},
+						},
+						"total_target_capacity": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"terminate_instances": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"terminate_instances_with_expiration": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  ec2.FleetTypeMaintain,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.FleetTypeMaintain,
+					ec2.FleetTypeRequest,
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceAwsEc2FleetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	input := &ec2.CreateFleetInput{
+		ExcessCapacityTerminationPolicy:  aws.String(d.Get("excess_capacity_termination_policy").(string)),
+		LaunchTemplateConfigs:            expandEc2FleetLaunchTemplateConfigRequests(d.Get("launch_template_configs").([]interface{})),
```

Comment:
I feel like these casts are panic potential, but won't hold up a merge over it

Reply:
Calling `d.Get()` against a `schema.TypeList` will always return a `[]interface{}` with 0 or more elements 😄